lib/zip_source_zip_new.c

```diff
@@ -126,6 +188,11 @@ zip_source_t *_zip_source_zip_new(zip_t *srcza, zip_uint64_t srcidx, zip_flags_t
             return NULL;
         }
     }
+    else {
+        if ((src = _zip_source_window_new(srcza->entry[srcidx].source, 0, data_len, &st, &attributes, NULL, 0, error)) == NULL) {
```

Comment:
Another reason for wrapping it into window source is to make it possible to open `srcza->entry[srcidx].source` multiple times via `zip_file_t` and read each of the `zip_file_t` independently, because window source does a seek before each read.

Reply:
We'll do this post merge.